joint_teapot/utils/joj3.py

```diff
@@ -1,14 +1,139 @@
+from __future__ import annotations
```

Comment:
functions like: `def init(self, text: str, url: str) -> Failed_Table.Row.Link:`, without this import `-> Failed_Table.Row.Link` would report error.

I'm using this since I need several different construction functions and python cannot do overload. If I write a normal method to init, I can not write codes like `self.rows[i].failure = self.Row.Link().init(failure_name, failure_link)`. Instead it would be
```python
failure = self.Row.Link()
failure.init(failure_name, failure_link)
self.rows[i].failure = failure
```
Making the method to return itself saves two lines